File: 3195-find-the-minimum-area-to-cover-all-ones-i/3195-find-the-minimum-area-to-cover-all-ones-i.cpp
class Solution {
public:
    int minimumArea(vector<vector<int>>& grid) {
        int minI=INT_MAX;int minJ=INT_MAX;
        int maxI=INT_MIN;int maxJ=INT_MIN;
        for(int i=0;i<grid.size();i++){
            for(int j=0;j<grid[0].size();j++){
                if(grid[i][j]==1){
                    minI=min(minI,i);
                    minJ=min(minJ,j);
                    maxI=max(maxI,i);
                    maxJ=max(maxJ,j);
                }
            }
        }
        return (maxI-minI+1)*(maxJ-minJ+1);
    }
};

File: 3195-find-the-minimum-area-to-cover-all-ones-i/3195-find-the-minimum-area-to-cover-all-ones-i.java
class Solution {
    public int minimumArea(int[][] grid) {
        int m = grid.length;
        int n = grid[0].length;
        int minHor=n;
        int maxHor=0;
        int minVer=m;
        int maxVer=0;
        for(int i=0;i<m;i++) {
            for(int j=0;j<n;j++) {
                if(grid[i][j]==1) {
                    minHor = Math.min(minHor,j);
                    minVer = Math.min(minVer,i);
                    maxHor = Math.max(maxHor,j);
                    maxVer = Math.max(maxVer,i);
                }
            }
        }

        return (maxHor-minHor+1)*(maxVer-minVer+1);
    }
}